### ISSUE TITLE
Add intersections to multiple curves

### ISSUE THREAD

**Issue by JanFilipec** [nortikin/sverchok#3599]
## Problem statement

I'm trying to insert nodes to points where curves intersect.

## Steps to reproduce

1. Input multiple curves
2. Search for intersections
3. Output original curves with new nodes on places where they intersect

## Expected result

original curves with new nodes on places where they intersect

## Actual result

Intersect edges works great, but I can't seem to be able to join the original curves together again or to insert the new points into the original curves.
Also, is it possible to merge vertices of two curves to join them when their endpoints touch?

**Comment by portnov** [comment]
Sorry, not quite clear what are you talking about. Maybe it would be simpler to illustrate with screenshots or schematic drawings.

**Comment by portnov** [comment]
> I'm trying to insert nodes to points where curves intersect.

Are you talking about adding control points to NURBS curves? or?...

**Comment by JanFilipec** [comment]
My input (only two curves for illustration, but I need many more)
![Input](https://user-images.githubusercontent.com/67052298/94606319-94d2e380-029a-11eb-8919-1159ddac8a8e.JPG)
Expected output-two separate curves with new nodes
![Result](https://user-images.githubusercontent.com/67052298/94606402-b338df00-029a-11eb-98c4-24ea3696cb79.JPG)
![Result 2](https://user-images.githubusercontent.com/67052298/94606409-b633cf80-029a-11eb-9571-c8b0b4bb1af1.JPG)
Right now I need only polylines (these are mesh edges), honestly I still don't quite get the system how sverchok handles the different types of curves or what is actually a curve in sverchok terms, is there a detailed documentation for this? Somehow when I draw a curve with blender and load it into sverchok I have problems with nodes that take curves as input, but I suppose I just don't understand how it works.

**Comment by portnov** [comment]
Aha, I see, so the question is mostly about how to apply "intersect edges" node in this case. 

About curves in Sverchok terms — see http://nikitron.cc.ua/sverch/html/curves.html

**Comment by JanFilipec** [comment]
Thanks, this I have seen already, however for example creating a bezier curve in blender, loading it with object id selector node and plugging it into curve length parameter gives me a red node.
Also how do I create a sverchok curve from individual points? Or convert a mesh edge into a curve?
And can I join curves when their endpoints touch? And merge mesh vertices?

**Comment by JanFilipec** [comment]
![Capture](https://user-images.githubusercontent.com/67052298/94610479-a4edc180-02a0-11eb-9904-93dee95af425.JPG)

**Comment by portnov** [comment]
> for example creating a bezier curve in blender, loading it with object id selector node and plugging it into curve length parameter gives me a red node.

"Object selector" node outputs Blender objects (`bpy.data....`). To load curves as Sverchok curves, there is "Bezier In" node for Bezier curves and "NURBS In" for NURBS curves. In most cases, you can easily recognize if output of one node is compatible with input of another, just by color of the socket.

> Also how do I create a sverchok curve from individual points? 

There are several nodes that build curves by interpolation of points: "Cubic spline", "Interpolating Nurbs curve", "RBF curve"... There are also nodes with "approximation" in name.

> Or convert a mesh edge into a curve?

"Polyline curve" or "Edge boom" + "Line (Curve)", depending on what exactly do you want.

> And can I join curves when their endpoints touch?

"Concatenate curves"

In short, you have to understand that "curves" in Sverchok is a completely different object, not having any relation to meshes; it can be created from mesh vertices by interpolation and approximation, and it can be converted to mesh by "evaluate curve" node. To manipulate with curves, there is a whole category of nodes. This is mostly similar to Grasshopper's approach (with only difference that in GH, curves and surfaces are primary objects, and  mesh is like... "say thank you that we support meshes at all"; while in Sverchok it's almost vice versa — most nodes are for working with mesh).

**Comment by JanFilipec** [comment]
Ok great, I played with all these nodes a bit and I think I understand it much better now, thanks a lot.
Back to my original question, do you have any idea how to insert the intersections vertices with mesh edges or curves?

**Comment by rendetto** [comment]
There is no direct method (or node) that can do that as far as I know.  I'm sure that it's doable with the amount of tools available but you'll need to build more or less a complex logic tree that finds the intersections and then use those to cut the original curves. You can find intersection points for mesh edges using either the "Split edges" or the "Intersect Edges" nodes.

Cheers

**Comment by JanFilipec** [comment]
I used "Intersect edges" first, masked out new vertices, found which curve they belong to with "Distance point-line", inserted them with "List item insert" and made new edges with "KDT closest path". Not simple, but it works with sverchok tools only. 
Thanks for help.
